dvc/command/stage.py

```diff
@@ -194,3 +292,41 @@ def add_parser(subparsers, parent_parser):
     )
     _add_common_args(stage_add_parser)
     stage_add_parser.set_defaults(func=CmdStageAdd)
+
+    STAGE_LIST_HELP = "List stages."
+    stage_list_parser = stage_subparsers.add_parser(
+        "list",
+        parents=[parent_parser],
+        description=append_doc_link(STAGE_LIST_HELP, "stage/list"),
+        help=STAGE_LIST_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    stage_list_parser.add_argument(
+        "targets", nargs="*", default=["dvc.yaml"], help="Stages to list from",
```

Comment:
Files to list from?

Reply:
@shcheklein, it can also be a directory/foreach-stage-name/stage-name.

Reply:
kk, my point that this combination `default=["dvc.yaml"]` and `"Stages to list from"` sounds strange a bit and it's not clear what does it exactly expect/can take.

Reply:
This has been a common problem in most of our commands these days as they all support these kinds of target spec. I don't have a good idea to make it concise and also document it clearly.

I changed it to the following (removed `or a stage name` to keep it _short_ for now):
> Show stages from a `dvc.yaml`/`.dvc` file or a directory. 'dvc.yaml' by default.

I looked into `pip install` and `git add`, and they mostly seem to not explain positional arguments.

1. `pip`
```console
$ pip install -h

Usage:   
  pip install [options] <requirement specifier> [package-index-options] ...
  pip install [options] -r <requirements file> [package-index-options] ...
  pip install [options] [-e] <vcs project url> ...
  pip install [options] [-e] <local project path> ...
  pip install [options] <archive url/path> ...
```

2. `git`
```console
$ git add -h
usage: git add [<options>] [--] <pathspec>...
```

maybe we could do similar to pip in our commands?

Reply:
You are right, but in this case it should be less generic as `targets`. In case of pip `local project path` is a good example. It doesn't require any further explanations.

Git is so-so. They use `pathspec` but they are very consistent and it's easy to google it (it's very specific). Also, they have examples usually around.

Reply:
We could also replace similar to the pip with:
```console
$ dvc stage list -h

Usage:
	dvc stage list [path to dvc.yaml]...
	dvc stage list -R <dir>...
	dvc stage list [path to dvc.yaml][:<stage_name>]...
	dvc stage list --all
```